gobltin.go

```diff
@@ -0,0 +1,37 @@
+// Package gobltin implements the lookup of TIN numbers
+package gobltin
+
+import (
+	"context"
+
+	"github.com/invopop/gobl/org"
+)
+
+// LookupTin checks the validity of the TIN number for the customer and/or the supplier in an invoice.
+// The function returns a list of PartyTinResponse, one for each party type requested.
+//
+// The user can choose to validate the customer, the supplier, or both, included as an argument.
+// If the party type is not provided, the function will validate only the customer.
+func LookupTin(ctx context.Context, party *org.Party) (bool, error) {
```

Comment:
I'd remove the `Tin` part of the name as we already know its a "TIN" from the package name. Also, Go convention suggests that abbreviations should be capitalised, so the correct method name would be `LookupTIN`. (I'd go with just `Lookup` though).

You might always want to consider making this method more generic. You could accept `any` instead of `org.Party` then try to work out what you can check based on the type.

Reply:
At first, I had done it for an invoice as input, but in that case there should also be another argument that is if you want to validate the customer, the supplier or both. What is more, if it is an invoice and you are validating both, the function will return an array specifying which one is the customer and the supplier. This is what I had before and I believed it was added complexity when the user can just do two calls to the function when they need the customer and the supplier.

Reply:
I agree in any case that it would be useful to have an any for the case the user just wants to check the id and not the org.Party